### ISSUE TITLE
test

### ISSUE THREAD

**Issue by sosizk** [sosizk/miniAI_Fruit#2]
import numpy as np
import matplotlib.pyplot as plt
import tensorflow as tf
from tensorflow.keras.preprocessing.image import ImageDataGenerator
from tensorflow.keras.applications import MobileNetV2
from tensorflow.keras import layers, models
import os

IMG_SIZE = (224, 224)
BATCH_SIZE = 32
EPOCHS = 10
DATASET_DIR = r"C:\Users\User\Downloads\miniproject\fruit\train" 

print("กำลังเตรียมข้อมูลและทำ Image Augmentation...")
datagen = ImageDataGenerator(
    rescale=1./255,
    rotation_range=30,
    width_shift_range=0.2,
    height_shift_range=0.2,
    shear_range=0.2,
    zoom_range=0.2,
    horizontal_flip=True,
    fill_mode='nearest',
    validation_split=0.3
)

train_generator = datagen.flow_from_directory(
    DATASET_DIR,
    target_size=IMG_SIZE,
    batch_size=BATCH_SIZE,
    class_mode='categorical', 
    subset='training'
)

val_generator = datagen.flow_from_directory(
    DATASET_DIR,
    target_size=IMG_SIZE,
    batch_size=BATCH_SIZE,
    class_mode='categorical',
    subset='validation'
)

class_indices = train_generator.class_indices
num_classes = len(class_indices)
print(f"พบจำนวนคลาสทั้งหมด: {num_classes} คลาส")
print(f"การจำแนกคลาส: {class_indices}")

print("\nกำลังสร้างโมเดลด้วย MobileNetV2 (Transfer Learning)...")
base_model = MobileNetV2(weights='imagenet', include_top=False, input_shape=(224, 224, 3))
base_model.trainable = False 

model = models.Sequential([
    base_model,
    layers.GlobalAveragePooling2D(),
    layers.Dense(128, activation='relu'),
    layers.Dropout(0.3),
    layers.Dense(num_classes, activation='softmax') 
])

model.compile(
    optimizer=tf.keras.optimizers.Adam(learning_rate=0.0001),
    loss='categorical_crossentropy', 
    metrics=['accuracy']
)

print("\nเริ่มทำการเทรนโมเดล...")
history = model.fit(train_generator, validation_data=val_generator, epochs=EPOCHS)

model.save('fruit_multi_class_model.h5')
print("\nบันทึกโมเดลสำเร็จในชื่อ 'fruit_multi_class_model.h5'")

acc = history.history['accuracy']
val_acc = history.history['val_accuracy']
loss = history.history['loss']
val_loss = history.history['val_loss']
epochs_range = range(1, EPOCHS + 1)

plt.figure(figsize=(14, 5))
plt.subplot(1, 2, 1)
plt.plot(epochs_range, acc, 'b-', label='Training Accuracy', linewidth=2)
plt.plot(epochs_range, val_acc, 'g--', label='Validation Accuracy', linewidth=2)
plt.title('Accuracy')
plt.legend()
plt.grid(True, linestyle='--', alpha=0.7)

plt.subplot(1, 2, 2)
plt.plot(epochs_range, loss, 'r-', label='Training Loss', linewidth=2)
plt.plot(epochs_range, val_loss, 'orange', linestyle='--', label='Validation Loss', linewidth=2)
plt.title('Loss')
plt.legend()
plt.grid(True, linestyle='--', alpha=0.7)

plt.tight_layout()
plt.savefig('multi_class_graph.png', dpi=300) 
plt.show()
